plugins/trace_api_plugin/include/eosio/trace_api/compressed_file.hpp

```diff
@@ -0,0 +1,87 @@
+#pragma once
+
+#include <ios>
+#include <fc/io/cfile.hpp>
+
+namespace eosio::trace_api {
+
+   class compressed_file_datastream;
+   struct compressed_file_impl;
+   /**
+    * wrapper for read-only access to a compressed file.
+    * compressed files support seeking though the efficiency of random access is lower than
+    * an uncompressed file in trade for smaller on-disk resource usage
+    */
+   class compressed_file {
+   public:
+      compressed_file( fc::path file_path );
+      ~compressed_file();
+
+      void open() {
+         file_ptr = std::make_unique<fc::cfile>();
+         file_ptr->set_file_path(file_path);
+         file_ptr->open("rb");
+      }
+
+      bool is_open() const { return (bool)file_ptr; }
+
+      void seek( long loc );
+
+      void read( char* d, size_t n );
+
+      void close() {
+         file_ptr.reset();
+      }
+
+      auto get_file_path() const {
+         return file_path;
+      }
+
+      compressed_file_datastream create_datastream();
+      
+      static bool process( const fc::path& input_path, const fc::path& output_path, uint16_t seek_point_count );
```

Comment:
Needs a comment indicating this converts the input_path file to a compressed file at output_path, and maybe a little more info with it.